django_project/base/models.py

```diff
@@ -208,3 +208,49 @@ def save_user_profile(sender, instance, **kwargs):
     Signal to save the UserProfile whenever the User is saved.
     """
     instance.profile.save()
+
+
+class UserOrganisations(models.Model):
+    """
+    Model to represent the relationship between a user and
+    an organisation with their respective role.
+    """
+    USER_TYPES = [
+        ('manager', 'Manager'),
+        ('member', 'Member'),
+    ]
+
+    user = models.ForeignKey(
+        User,
+        on_delete=models.CASCADE,
+        related_name="user_organisations",
+    )
+
+    user_profile = models.ForeignKey(
```

Comment:
I think we can use either user or user_profile in the UserOrganisations model because 1 user will only have 1 UserProfile. Can we use just 1 field either user or user_profile?

Reply:
the code is relying on both , when saving the organisation on the user profile and when querying for user organisations .it simplifies the process

Reply:
@tinashechiraya let's use 1 field (I prefer user field). This is to ensure data integrity, for example someone updates user field with user X, but user_profile is from user Y, then there will not be integrity.

Reply:
![Screenshot from 2024-12-06 17-41-11](https://github.com/user-attachments/assets/3fff7493-2798-48c9-87b9-a022c58dc21b)


@danangmassandy this is error i get when i remove user profiel

Reply:
it's because there is organisations field on UserProfile, you could try removing the user instead. But, we need to set the user_profile field as non-nullable